frontend/src/metabase/dashboard/actions/trash.ts

```diff
@@ -0,0 +1,101 @@
+import { t } from "ttag";
+import _ from "underscore";
+
+import { canonicalCollectionId } from "metabase/collections/utils";
+import Dashboards from "metabase/entities/dashboards";
+import { createThunkAction } from "metabase/lib/redux";
+import { addUndo } from "metabase/redux/undo";
+
+import { fetchDashboard } from "./data-fetching";
+
+// just using the entity action doesn't cause the dashboard to live update
+// calling fetchDashboard ensures that the view updates with the last values
+export const SET_ARCHIVED_DASHBOARD =
+  "metabase/dashboard/SET_ARCHIVED_DASHBOARD";
+export const setArchivedDashboard = createThunkAction(
+  SET_ARCHIVED_DASHBOARD,
+  function (archived = true) {
+    return async function (dispatch, getState) {
+      const { dashboardId } = getState().dashboard;
+
+      await dispatch(
+        Dashboards.actions.update({ id: dashboardId }, { archived }),
+      );
+
+      dispatch(
+        addUndo({
+          subject: t`dashboard`,
+          verb: archived ? t`trashed` : t`restored`,
+          action: () => dispatch(setArchivedDashboard(!archived)),
+        }),
+      );
+
+      dispatch(
+        // @ts-expect-error rtk typings are wrong - function expects an arguement but TS errors saying it shouldn't have one
```

Comment:
```suggestion
        // @ts-expect-error rtk typings are wrong - function expects an argument but TS errors saying it shouldn't have one
```

Should the string that identifies the specific error be included here?